runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1196,42 +1196,54 @@ end
 ---             - Exclusive severity to consider. Overrides {severity_limit}
 ---         - {severity_limit}: (DiagnosticSeverity)
 ---             - Limit severity of diagnostics found. E.g. "Warning" means { "Error", "Warning" } will be valid.
+---         - {workspace}: (boolean, default false)
+---             - Set the list with workspace diagnostics
 function M.set_loclist(opts)
   opts = opts or {}
 
   local open_loclist = if_nil(opts.open_loclist, true)
 
-  local bufnr = vim.api.nvim_get_current_buf()
-  local buffer_diags = M.get(bufnr, opts.client_id)
-
-  if opts.severity then
-    buffer_diags = filter_to_severity_limit(opts.severity, buffer_diags)
-  elseif opts.severity_limit then
-    buffer_diags = filter_by_severity_limit(opts.severity_limit, buffer_diags)
-  end
+  local current_bufnr = vim.api.nvim_get_current_buf()
+  local diags = opts.workspace and M.get_all() or {
+    [current_bufnr] = M.get(current_bufnr, opts.client_id)
+  }
 
   local items = {}
-  local insert_diag = function(diag)
+  local insert_diag = function(bufnr, diag)
     local pos = diag.range.start
     local row = pos.line
-    local col = util.character_offset(bufnr, row, pos.character)
 
     local line = (api.nvim_buf_get_lines(bufnr, row, row + 1, false) or {""})[1]
+    local col = line and util.character_offset(bufnr, row, pos.character) or 0
```

Comment:
Why are you testing for validity of line here? When would this not be valid?

Reply:
I noticed that sometimes nvim_buf_get_lines (which is also used in character_offset) would return a null value. Maybe because the buffer isn't loaded? (no hidden is set) I don't really know the root cause, but line can be null

Reply:
I'd rather catch the fault where it occurs then add this workaround...

Reply:
No, not a comment, check if the buffer is valid before calling buf_get_lines.

Reply:
Alright, I changed util.character_offset to return 0 on a null line. It looks like other places in util.lua also guard against null lines, just not this function.

Reply:
I don't think that's a good idea, I'm saying check if the buffer is a valid buffer before calling any of the functions, we don't want to invent an error code system that's applied arbitrarily to different functions.

Reply:
@mjlbach 

How do I go about checking if the buffer is valid? `vim.fn.bufexists`?

Reply:
@mjlbach 

I did some experiments calling every possibly buf* function I could think of when the line is null:

```
{
  col = 20,
  exists = 1,
  listed = 0,
  load = 0,
  loaded = 1,
  name = "some/file_test.go",
  row = 70
}
```

That file has content on line 71 col 21, so I'm not sure why nvim_buf_get_lines returns null

Reply:
Most of the buffers seem to return null here. The only consequence of that with this change will be that the list will not include the relevant text from the buffer, but will still include the description.

Reply:
it looks like it will return null for any unloaded buffer, which all except the ones currently visible will be.

Reply:
@mjlbach 

I'm now checking whether a buffer is loaded before the character offset and line is being fetched.